### ISSUE TITLE
Adding tagged releases to Docker Hub

### ISSUE THREAD

**Issue by ruimarinho** [clkao/docker-postgres-plv8#12]
@clkao I think it's important to start tagging releases on github and auto building tags on Docker Hub so that users can pull the tagged release `docker pull postgres-plv8:1.5.2` instead of latest.

WDYT?

**Comment by clkao** [comment]
+1 on having plv8 point release tags on docker hub, but i think we still want to support different pg versions?

not sure about tagging in github though - doesn't seem to be what https://github.com/docker-library/postgres is doing. is there a docker source repo using tags?

**Comment by ruimarinho** [comment]
I guess those repositories haven't yet moved to the new system provided by Docker Hub, which lets you match git tags to build tags. 

I was thinking something where Docker tagging looks like this: `postgres-plv8:9.5-1.4.8`, `postgres-plv8:9.5-1.5.3`, `postgres-plv8:9.4-1.4.8`, `postgres-plv8:9.4-1.5.3`.

The git counterpart tag would simply be the plv8 version for now (`1.4.8`), as we can change the Docker tag build name dynamically. This works because plv8 r1.4 and r1.5 both compile for postgres 9.4 and 9.5. When r2 is out, we will have to adjust the tag regex, but I think that's fine.

Docker Hub config (cannot add it myself):

```
Type: Tag
Name: ^1\.5\.[0-9.]+$
Dockerfile Location: /9.5-1.5
Docker Tag Name: 9.5-{sourceref}
```

```
Type: Tag
Name: ^1\.5\.[0-9.]+$
Dockerfile Location: /9.4-1.5
Docker Tag Name: 9.4-{sourceref}
```

```
Type: Tag
Name: ^1\.4\.[0-9.]+$
Dockerfile Location: /9.5-1.4
Docker Tag Name: 9.5-{sourceref}
```

```
Type: Tag
Name: ^1\.4\.[0-9.]+$
Dockerfile Location: /9.4-1.4
Docker Tag Name: 9.4-{sourceref}
```

**Comment by ruimarinho** [comment]
Any luck with these @clkao? I'm not sure how Docker Hub handles external collaborators, but I can help with this if you want and Docker Hub allows.

**Comment by ruimarinho** [comment]
Got any update on this @clkao?

**Comment by ruimarinho** [comment]
@clkao sorry to be a PITA, but do you have time to take a look at this soon?

**Comment by clkao** [comment]
@ruimarinho Hi! sorry i've been busy.  Looking into this now.

**Comment by clkao** [comment]
it feels a bit painful that any logical change will go into multiple tags, and having tags not being immutable feels a bit icky.  Unless there's a good way to programatically generate tag contents from a template (perhaps out of master), I think we should make the repo available as it is for the sake of clarify.  also the [postgis](https://github.com/appropriate/docker-postgis) repo is havin the similar layout.

**Comment by ruimarinho** [comment]
@clkao indeed, it has that side affect but that will only affect the `latest` tag of each branch (`9.X-1.X`), which is considered mutable by default. All others depending on an explicit tag remain as static as we want them to be. It's hard to make mistakes with it, even if the setup is a bit convoluted.
